src/utils/seo/images.ts

```diff
@@ -0,0 +1,267 @@
+  getFonts,
+  getAssetPath,
+  getImage,
+  getSnapsWithIcon,
+} from './utils';
+
+/**
+ * Create a fallback icon for a Snap. This will create an icon with a default
+ * background and the provided text.
+ *
+ * @param text - The text to use for the icon.
+ * @param size - The size of the icon. Defaults to 256.
+ * @param fontPath - The path to the font to use. Defaults to
+ * 'fonts/icon/grey.fnt'.
+ */
+export async function createFallbackIcon(
+  text: string,
+  size = 256,
+  fontPath = 'fonts/icon/grey.fnt',
+) {
+  const image = await Jimp.create(size, size);
+  const font = await Jimp.loadFont(getAssetPath(fontPath));
+
+  const letterX = (size - Jimp.measureText(font, text)) / 2;
+  const letterY = (size - Jimp.measureTextHeight(font, text, size)) / 2;
+
+  image.print(font, letterX, letterY, text);
+
+  const icon = getImage('icon');
+  const input = await image.getBufferAsync(Jimp.MIME_PNG);
+
+  return icon
+    .resize(size, size)
+    .composite([
+      {
+        input,
+        top: 0,
+        left: 0,
+      },
+    ])
+    .png()
+    .toBuffer();
+}
+
+/**
+ * Create a circular icon for a Snap. If the Snap doesn't have an icon, an icon
+ * with the first letter of the name will be created.
+ *
+ * @param name - The name of the Snap.
+ * @param svgIcon - The SVG icon of the Snap.
+ * @param size - The size of the icon. Defaults to 256.
+ */
+export async function createSnapIcon(
+  name: string,
+  svgIcon?: string,
+  size = DEFAULT_ICON_SIZE,
+) {
+  if (svgIcon) {
+    const normalisedSvgIcon = decodeURIComponent(
+      svgIcon.replace(/data:image\/svg\+xml;utf8,/gu, ''),
+    );
+
+    const background = getImage('flat-icon').resize(size, size);
+    const icon = sharp(Buffer.from(normalisedSvgIcon, 'utf-8')).resize(
+      size,
+      size,
+    );
+
+    // Mask the icon with the mask image to create a circular icon.
+    const mask = await getImage('mask').resize(size, size).png().toBuffer();
+    const circularIcon = await icon
+      .composite([
+        {
+          input: mask,
+          blend: 'dest-in',
+        },
+      ])
+      .png()
+      .toBuffer();
+
+    return await background
+      .composite([
+        {
+          input: circularIcon,
+          top: 0,
+          left: 0,
+        },
+      ])
+      .png()
+      .toBuffer();
+  }
+
+  const letter = name.charAt(0).toUpperCase();
+  return createFallbackIcon(letter, size);
+}
+
+const SNAP_ICON_OFFSET_X = 239;
+const SNAP_ICON_OFFSET_Y = 78;
+
+const SNAP_COUNT_X = SNAP_ICON_OFFSET_X + 112 * 5;
+const SNAP_COUNT_Y = SNAP_ICON_OFFSET_Y;
+
+/**
+ * Create a PNG image for a category.
+ *
+ * @param snaps - The snaps to create the category image for.
+ * @returns The buffer of the image.
+ */
+export async function createCategoryImage(snaps: Snap[]) {
+  const filteredSnaps = getSnapsWithIcon(snaps);
+  const randomSnaps = shuffle(filteredSnaps).slice(0, 5);
+
+  const snapsLength = filteredSnaps.length > 5 ? 6 : filteredSnaps.length;
```

Comment:
```suggestion
  const snapsLength = Math.min(6, filteredSnaps.length);
```
Perhaps slightly more readable?